### ISSUE TITLE
Expose configuration options as Terraform attributes instead of passing in YAML heredoc

### ISSUE THREAD

**Issue by jrhouston** [kyma-incubator/terraform-provider-kind#13]
👋 Thanks for making a provider for kind, this is super helpful for me. 

## Description

Something I do with some frequency is create kind clusters with certain feature gates enabled. I'd like to be able to do this just with attributes on the resource, instead of having to pass in a YAML heredoc.

There's a couple of ways we could do this. The ideal way would be to codify the schema for the cluster config file into the kind resource itself, so we could do things like this:

### Example

```hcl
resource "kind "example" {
   name = "example"
  
   feature_gates = {
       StartupProbe = true,
       TokenRequest = false,
   }
}
```

An advantage of this is that `terraform validate` will catch more problems upfront, but the downside is the maintenance burden of keeping the schema up to date. I haven't checked but if kind publishes a schema for their config file format we could probably automate this.

A simpler way of doing it would be to expose a `config` attribute that accepts an arbitrary map, which the provider could then decode into YAML, e.g:

```hcl
resource "kind" "example" {
  name = "example"
  config = {
     kind = "Cluster"
     apiVersion = "kind.x-k8s.io/v1alpha4"
     featureGates = {
       TokenRequest = true
     }
  }
}
```

This would be easy to implement but suffers the problem that if you mess up the config format you wont find out until it blows up when you run apply.

I'm going to take a stab at this myself, but I wanted to open an issue for it capture my thoughts.

**Comment by tehcyx** [comment]
Hey John,
this sounds indeed very interesting. Ben from the kind team mentioned in the past that the API for kind is not very stable. I haven't recently caught up if that statement changed and therefor would not advise for maintaining a copy of the schema in this repository. A looser coupling is probably a better fit for this, like something you outlined in your second example.

Let me know if you need anything from our side.

**Comment by tehcyx** [comment]
Merged this yesterday in #26
